chainer/functions/array/permutate.py

```diff
@@ -0,0 +1,89 @@
+import numpy
+import six
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+def _check_indices(indices):
+    if len(indices) == 0:
+        return
+    for i in indices:
+        if 0 <= i < len(indices):
+            continue
+        raise ValueError('Out of bounds index: {}'.format(i))
+    sort = numpy.sort(indices)
+    for s, t in six.moves.zip(sort, sort[1:]):
+        if s == t:
+            raise ValueError('indices contains duplicate value: {}'.format(s))
+
+
+def _reverse_indices(indices):
```

Comment:
This operation is to take inverse of the permutation, rather than reverse of them.